packages/shared/src/lib/core/wallet/actions/send/createEvmChainToEvmChainTransaction.ts

```diff
@@ -77,47 +83,43 @@ export function createEvmChainToEvmChainTransaction(
 }
 
-function getDataForTransaction(
+function getTokenDataForTransaction(
     chain: IChain,
-    originAddress: string,
     recipientAddress: string,
-    token: IToken | undefined,
-    amount: string | undefined,
-    nft: Nft | undefined
+    token: IToken,
+    amount: string
 ): string | undefined {
-    if (token && amount) {
-        const standard = token.metadata?.standard
-        switch (standard) {
-            case TokenStandard.Irc30: {
-                const isBaseCoin = token.standard === TokenStandard.BaseToken
-                const assetType = isBaseCoin ? AssetType.BaseCoin : AssetType.Token
-                const transferredAsset = { type: assetType, token, amount } as TransferredAsset
-                return getIscpTransferSmartContractData(recipientAddress, transferredAsset, chain)
-            }
-            case TokenStandard.Erc20:
-                return getErc20TransferSmartContractData(recipientAddress, token, amount, chain)
-            default:
-                return undefined
+    switch (token?.metadata?.standard) {
+        case TokenStandard.Irc30: {
+            const isBaseCoin = token.standard === TokenStandard.BaseToken
+            const assetType = isBaseCoin ? AssetType.BaseCoin : AssetType.Token
+            const transferredAsset = { type: assetType, token, amount } as TransferredAsset
+            return getIscpTransferSmartContractData(recipientAddress, transferredAsset, chain)
         }
-    } else if (nft) {
-        const transferredAsset = { type: AssetType.Nft, nft } as TransferredAsset
-        switch (nft.standard) {
-            case NftStandard.Irc27:
-                return getIscpTransferSmartContractData(recipientAddress, transferredAsset, chain)
-            case NftStandard.Erc721: {
-                const nftAddress = nft?.contractMetadata?.address ?? ''
-                const nftTokenId = nft?.tokenId ?? ''
-                return getErc721TransferSmartContractData(
-                    originAddress,
-                    recipientAddress,
-                    nftAddress,
-                    nftTokenId,
-                    chain
-                )
-            }
-            default:
-                return undefined
+        case TokenStandard.Erc20:
+            return getErc20TransferSmartContractData(recipientAddress, token, amount ?? '', chain)
+        default:
+            return undefined
+    }
+}
+
+function getNftDataForTransaction(
+    chain: IChain,
+    originAddress: string,
+    recipientAddress: string,
+    nft: Nft
+): string | undefined {
+    const transferredAsset = { type: AssetType.Nft, nft } as TransferredAsset
+    switch (nft.standard) {
+        case NftStandard.Irc27:
+            return getIscpTransferSmartContractData(recipientAddress, transferredAsset, chain)
+        case NftStandard.Erc721: {
+            const nftAddress = nft.contractMetadata?.address ?? ''
+            const nftTokenId = nft.tokenId ?? ''
```

Comment:
Move this part into `getErc721TransferSmartContractData`, so you don't have as many variables.